app/js/lib/store_update_handling.js

```diff
@@ -50,5 +50,27 @@ module.exports = {
 				);
 			} );
 		} );
+	},
+	makeEventHandlerWaitForAsyncFinish: function ( handler, store ) {
+		return function () {
+			var unsubscribe = null;
+
+			if ( !store.getState().asynchronousRequests.isValidating ) {
+				handler();
+				return;
+			}
+
+			if ( unsubscribe !== null ) {
```

Comment:
No, it's the anonymous function in lines 66-73. That function checks if all asynchronous requests have finished and if they did, the handler and `unsubscribe` get executed and `unsubscribe` is unset.